File: 2-add-two-numbers/add-two-numbers.py
# Definition for singly-linked list.
# class ListNode(object):
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution(object):
    def addTwoNumbers(self, l1, l2):
        """
        :type l1: Optional[ListNode]
        :type l2: Optional[ListNode]
        :rtype: Optional[ListNode]
        """
        result= ListNode(0);
        carry=0
        current=result

        while(l1 is not None or l2 is not None):
            val1=0
            val2=0
            if(l1 is not None):
                val1=l1.val
            else:
                val1=0
            if(l2 is not None):
                val2=l2.val
            else:
                val2=0
            sum=val1+val2+carry

            carry=sum//10
            temp=sum%10
            current.next=ListNode(temp)
            current=current.next

            if(l1!=None):
                l1=l1.next
            if(l2!=None):
                l2=l2.next

        if(carry>0):
            current.next=ListNode(carry)
        return result.next
        

File: 2-add-two-numbers/add-two-numbers.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public ListNode addTwoNumbers(ListNode l1, ListNode l2) {
        int carry=0;
        ListNode result=new ListNode(0);
        ListNode current=result;

        while(l1!=null || l2!=null){
            int val1=(l1!=null)?l1.val:0;
            int val2=(l2!=null)?l2.val:0;
            int sum=val1+val2+carry;
            carry=sum/10;
            int temp =sum%10;

            current.next=new ListNode(temp);
            current=current.next;
            if(l1!=null) l1=l1.next;
            if(l2!=null)  l2=l2.next;
            
        }
        if(carry>0){
            current.next=new ListNode(carry);
        }
        return result.next;
    }
}